cmake/SetupPackages.cmake

```diff
@@ -81,3 +81,35 @@ endif ()
 if (CALIPER_FOUND)
     set(RT_LIBRARIES "${RT_LIBRARIES} ${caliper_LIB_DIR}/libcaliper.so" CACHE STRING "testing libraries" FORCE)
 endif()
+
+if (RAJA_ENABLE_TESTS)
+  include(ExternalProject)
+  # Set default ExternalProject root directory
+  SET_DIRECTORY_PROPERTIES(PROPERTIES EP_PREFIX ${CMAKE_BINARY_DIR}/tpl)
+
+  ExternalProject_Add(
+      googletest
+      GIT_REPOSITORY https://github.com/google/googletest.git
+      GIT_TAG release-1.7.0
+      INSTALL_COMMAND ""
+      LOG_DOWNLOAD ON
+      LOG_CONFIGURE ON
+      LOG_BUILD ON)
+
+  ExternalProject_Get_Property(googletest source_dir)
+  include_directories(${source_dir}/include)
+
+  ExternalProject_Get_Property(googletest binary_dir)
+  add_library(gtest      UNKNOWN IMPORTED)
+  add_library(gtest_main UNKNOWN IMPORTED)
+  set_target_properties(gtest PROPERTIES
+    IMPORTED_LOCATION ${binary_dir}/libgtest.a
+  )
+  set_target_properties(gtest_main PROPERTIES
+    IMPORTED_LOCATION ${binary_dir}/libgtest_main.a
+  )
+  add_dependencies(gtest      googletest)
```

Comment:
I take that back. Looking at it now, it looks like pthreads isn't getting linked into the tests that don't actually use openmp or something.  Since gtest uses the pthread_{get,set}specific interface for thread-local state, it has to have it even for serial tests.